hardware/raspberrypi/booteeprom.md

```diff
@@ -2,82 +2,118 @@
 ```
-sudo rpi-eeprom-update -a
+sudo rpi-eeprom-config --apply boot.conf
 sudo reboot
 ```
 
-### Reading the current EEPROM configuration
+### Reading the configuration file from an EEPROM image
+To read the configuration file from an EEPROM image file:
+```
+rpi-eeprom-config pieeprom.bin
+```
+
+### Low level commands
 
-To view the configuration file used by the bootloader at boot time
+#### Updating the configuration in an EEPROM image
+The following command reads `pieeprom.bin` and replaces the configuration file with the contents of `boot.conf`. The result is written to `new.bin`
 ```
-vcgencmd bootloader_config
+rpi-eeprom-config --config boot.conf --out new.bin pieeprom.bin
 ```
 
-### Reading the EEPROM version
+For more information about advanced options please run `rpi-eeprom-config -h`.
+
+#### Updating the bootloader EEPROM
+The following will cause the bootloader EEPROM to be updated the next time the system is rebooted.
 ```
-vcgencmd bootloader_version
+# -d instructs rpi-eeprom-update to use the EEPROM configuration inside new.bin instead of
+# of replacing it with the current configuration from vcgencmd bootloader.
+sudo rpi-eeprom-update -d -f new.bin
 ```
+For more information about advanced options please run `rpi-eeprom-update -h`.
+
+### recovery.bin
+At power on, the BCM2711 ROM looks for a file called `recovery.bin` in the root directory of the boot partition on the sd-card. If a valid `recovery.bin` is found then the ROM executes this instead of the SPI EEPROM image. This mechanism ensures that the bootloader SPI EEPROM can always be reset to valid image with factory default settings.
+
+#### EEPROM update files
+| Filename | Purpose |
+|----------|---------|
+| recovery.bin | VideoCore EEPROM recovery executable |
+| pieeprom.upd | Bootloader EEPROM image |
+| pieeprom.bin | Bootloader EEPROM image (same as pieeprom.upd) |
+| pieeprom.sig | The sha256 checksum of bootloader image (pieeprom.upd/pieeprom.bin) |
+| vl805.bin | The VLI805 USB firmware EEPROM image. Pi 4 revision 1.4 and newer do not have a dedicated SPI EEPROM for the VL805 USB controller |
+| vl805.sig | The sha256 checksum of vl805.bin |
+
+* If the bootloader update image is called `pieeprom.upd` then `recovery.bin` renames itself to `recovery.000` and resets the CPU. Since `recovery.bin` is no longer present the ROM loads the newly bootloader from SPI EEPROM and the OS is booted as normal.
+* The `.sig` files should just contain the sha256 checksum (in hex) of the corresponding image file. Other fields should may be added in the future.
+* If the bootloader update image is called `pieeprom.bin` the `recovery.bin` will stop after the update has completed. On success the HDMI output will be green and the green activity LED is flashed rapidly. Otherwise, the HDMI output will be read and an error code will be displayed via the activity LED.
```

Comment:
the `recovery.bin` -> then `recovery.bin`

will be read -> will be red ?

Reply:
done